src/Widgets/Proxy/ProxyPage.vala

```diff
@@ -53,21 +77,140 @@ namespace Network.Widgets {
             proxy_settings.changed.connect (update_mode);
             update_mode ();
 
+#if ENABLE_SYSTEMWIDE_PROXY
+            permission_infobar = new Gtk.InfoBar ();
+            permission_infobar.message_type = Gtk.MessageType.INFO;
+
+            configuration_page.notify["manual-mode"].connect (() => {
+                update_infobar_visibility ();
+            });
+
+            var permission = get_permission ();
+
+            permission.notify["allowed"].connect (() => {
+                permission_infobar.visible = !permission.allowed;
+                system_wide_available = permission.allowed;
+            });
+
+            bind_property ("system-wide-available", configuration_page, "system-wide-available", BindingFlags.SYNC_CREATE);
+
+            var area_infobar = permission_infobar.get_action_area () as Gtk.Container;
+            var lock_button = new Gtk.LockButton (permission);
+            area_infobar.add (lock_button);
+
+            var content_infobar = permission_infobar.get_content_area () as Gtk.Container;
+            var label_infobar = new Gtk.Label (_("Administrator rights are required to set a system-wide proxy"));
+            content_infobar.add (label_infobar);
+#endif
+
+            var device_img = new Gtk.Image.from_icon_name ("preferences-system-network", Gtk.IconSize.DIALOG);
+            device_img.pixel_size = 48;
+
+            var device_label = new Gtk.Label (_("Proxy"));
+            device_label.ellipsize = Pango.EllipsizeMode.MIDDLE;
+            device_label.get_style_context ().add_class ("h2");
+            device_label.hexpand = true;
+            device_label.xalign = 0;
+
+            control_switch = new Gtk.Switch ();
+            control_switch.valign = Gtk.Align.CENTER;
+
+            control_switch.bind_property ("active", configuration_page, "sensitive", BindingFlags.SYNC_CREATE);
+            control_switch.bind_property ("active", exceptions_page, "sensitive", BindingFlags.SYNC_CREATE);
+            control_switch.notify["active"].connect (control_switch_activated);
+
+            var control_box = new Gtk.Grid ();
+            control_box.column_spacing = 12;
+            control_box.margin_left = control_box.margin_right = 24;
+            control_box.margin_top = 24;
+            control_box.add (device_img);
+            control_box.add (device_label);
+            control_box.add (control_switch);
+
+#if ENABLE_SYSTEMWIDE_PROXY
+            add (permission_infobar);
+#endif
+            add (control_box);
             add (stackswitcher);
             add (stack);
 
             show_all ();
 
             stack.visible_child = configuration_page;
+
+#if ENABLE_SYSTEMWIDE_PROXY
+            update_infobar_visibility ();
+#endif
+        }
+
+#if ENABLE_SYSTEMWIDE_PROXY
+        private void on_proxy_settings_changed () {
+            if (system_wide_available) {
```

Comment:
I don't like putting this entire block into one `if` check.
Can we do something like this for better readability?

```
if (!system_wide_available) {
    return;
}

...
```